internal/libs/wlog/wrappedlogger.go

```diff
@@ -0,0 +1,54 @@
+package wlog
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+type WrappedLogger struct {
+	baseLogger logger.Logger
+	data       map[string]any
+}
+
+func NewWrappedLogger(baseLogger logger.Logger) *WrappedLogger {
+	return &WrappedLogger{
+		baseLogger: baseLogger,
+		data:       make(map[string]any),
+	}
+}
+
+func (w *WrappedLogger) WithField(key string, value any) {
```

Comment:
I would love to be able to restrict this for only "simple" types like integers and strings. It would be great if, at a glance, I know the oracle ID and e.g, current state of the plugin.

```
[oracle 1] [state Filter] building report with 10 msgs
```

And in the log fields I have more details.

Then again, I think we could write some tools that do this kind of log analysis / transformation after pulling down the logs into a text file. The only issue with this is that it may not be possible due to grafana limitations etc.